pkg/apis/kudo/v1beta1/instance_types_helpers.go

```diff
@@ -196,6 +204,35 @@ func (i *Instance) IsTopLevelInstance() bool {
 	return !i.IsChildInstance()
 }
 
+const (
+	planInProgressReason   = "PlanInProgress"
+	planInFatalErrorReason = "PlanInFatalError"
+	resourceNotReadyReason = "ResourceNotReady"
+	resourcesReadyReason   = "ResourcesReady"
+	readyConditionType     = "Ready"
+)
+
+func (i *Instance) SetReadinessTrue(msg string) {
+	condition := metav1.Condition{Type: readyConditionType, Status: metav1.ConditionTrue, Message: msg, Reason: resourcesReadyReason}
+	meta.SetStatusCondition(&i.Status.Conditions, condition)
+}
+
+func (i *Instance) SetReadinessFalse(msg string) {
+	condition := metav1.Condition{Type: readyConditionType, Status: metav1.ConditionFalse, Message: msg, Reason: resourceNotReadyReason}
+	meta.SetStatusCondition(&i.Status.Conditions, condition)
+}
+
+func (i *Instance) SetReadinessFatalError(msg string) {
```

Comment:
Now, that `FATAL_ERROR` is also `Unknown` I would probably collapse this one into `func SetReadinessUnknown` and give the latter a reason as parameter.

Reply:
We would have to introduce an enum for `reason` in that case. Let's not use a string for that. But with an enum, we could go a step further and introduce `SetReadiness(status NewStatusEnum, msg string)` and have this function set `Status` and `Reason`. WDYT?

Reply:
Something doesn't fully add up here: Did you mean `SetReadiness(status StatusType, reason ReasonType, msg string)`?

We shouldn't add an enum but a custom type. Enums are problematic in the API as they define a complete set and can't easily be extended later on. A custom string type should be ok though...

Reply:
Custom string type is good as well. What I meant is that there are 4 statuses we want to cover: True, False, FatalError and PlanInProgress. What these imply for `StatusType` and `ReasonType` is, IMO, an implementation detail. So we can have a function `SetReadiness(status OurStatusType, msg string)` that then sets `StatusType`, `ReasonType` and the message for the status condition accordingly. Does that make sense?

Reply:
Isn't it the other way around? We have 4 Reasons and 3 StatusTypes, right? So, from the reason we can deduct the status, but not the other way around:
```
Reason | Status
ResourcesReady -> True
ResourceNotRead -> False
PlanInProgress -> Unknown
FatalError -> Unknown
```
So we would need `SetReadiness(reason ReasonType, msg string)`, right?

Reply:
Yes, good point. I implemented it like this now, though using `ReadinessType` instead of reason type so that it's clearer for callers. Under the hood this is then used to set the reason of `Condition`. PTAL.